### PR TITLE
Refactor work order data handler for WPE

### PR DESCRIPTION
**Pull request by manju956**
- Add separate work order data handler for WPE

Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: tc/sgx/trusted_worker_manager/enclave/work_order_data_handler_wpe.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "work_order_data_handler_wpe.h"
+
+namespace tcf {
+    void WorkOrderDataHandlerWPE::InitializeDataEncryptionKey() {
```
Comment (danintel) [inline tc/sgx/trusted_worker_manager/enclave/work_order_data_handler_wpe.cpp:19]: Add a short comment about what this is doing. Here and in the similar InitializeDataEncryptionKey() in file work_order_data_handler.cpp .
It looks like the data encryption key and IV are set to the session key and IV unless enc_data_key_str is "-", then they are set to empty.
In the latter case, does that mean there is no encryption?

Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/work_order_data_handler_wpe.cpp:19]: Yes Dan, if encryptedDataEncryptionKey is "-", data is sent from the requester as unencrypted text, hence there is no decryption needed for the data which is part of either inData or outData block in the work order request. 
I will mention the details in the comment.

File: tc/sgx/trusted_worker_manager/enclave/work_order_data_handler.cpp [hunk 2]
```diff
@@ -38,76 +38,38 @@
...
         } else if ("-" == enc_data_key_str) {
+            // If encrypted data encryption key is "-" in inData or outData
+            // data is unencrypted plain text. Hence setting data_encryption_key
+            // and data_iv set to empty values.
             data_encryption_key = {};
             data_iv = {};
         } else {
             data_iv = HexStringToBinary(iv);
-            ByteArray enc_data_key = HexStringToBinary(enc_data_key_str);
+            ByteArray enc_data_key_bytes = HexStringToBinary(enc_data_key_str);
```
[the remark this answers is not in the document]
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/work_order_data_handler.cpp:56]: > data_encryption_from_preprocessing

I will mention the double encryption details in the comment. Regarding the second suggestion, 
we already have GetDataEncryptionKey() function in WorkOrderDataHandler.h which returns data_encryption, a private member variable. 
So, i added a virtual function InitializeDataEncryptionKey() to WorkOrderDataHandler class and the same function is being overriden in WorkOrderDataHandlerWPE class.

[the remark this answers is not in the document]
Comment (manojgop) [reply tc/sgx/trusted_worker_manager/enclave/work_order_data_handler.cpp:56]: `InitializeDataEncryptionKey()` duplicates first 2 condition when `enc_data_key_str` is empty, null or "-".  We can avoid it. 
`GetDataEncryptionKey(ByteArray &data_enc_key, ByteArray &data_iv)` returns both key and iv.

[the remark this answers is not in the document]
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/work_order_data_handler.cpp]: Done

**Pull request merged**